src/get_shared_mem.cu

```diff
@@ -0,0 +1,24 @@
+#include <cuda_runtime.h>
+#include <cstdio>
+
+int main() {
+    int device = 0;
+    cudaDeviceProp prop;
+    cudaGetDeviceProperties(&prop, device);
+
+    size_t sharedMemPerBlock = prop.sharedMemPerBlock;
+    int maxSharedMemPerBlockOptin;
+    cudaDeviceGetAttribute(&maxSharedMemPerBlockOptin, cudaDevAttrMaxSharedMemoryPerBlockOptin, device);
+
+    size_t maxSharedMem = (sharedMemPerBlock > (size_t)maxSharedMemPerBlockOptin) ? sharedMemPerBlock : (size_t)maxSharedMemPerBlockOptin;
+
+    // Subtract 3KB (3072 bytes) from the max shared memory as is allocated somewhere else
+    maxSharedMem -= 3072;
```

Comment:
How did you compute this? I think this is from the automatic variables in some of the kernels. I think overall this is a fine approach for now but if we change the 3KB alloc in the runtime this will have to change too.

Reply:
the sum of the alloc size of some local shared arrays is roughly ~3KB, i know that this is not the perfect way to do this, but it works for now.